### ISSUE TITLE
Date picker documentation page fails Lighthouse checks

### ISSUE THREAD

**Issue by chosak** [cfpb/design-system#2124]
**Describe the bug**
The date picker page fails Lighthouse checks due to a missing heading `h2` element.

See nightly GitHub Action failure [here](https://github.com/cfpb/design-system/actions/runs/11963772970/job/33354848080) and failing Lighthouse report [here](https://storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/1732234794409-61114.report.html).

**To Reproduce**
1. Go to https://cfpb.github.io/design-system/components/date-picker.
2. Run Lighthouse check in Chrome.
3. See error.

**Screenshots**
<img width="745" alt="image" src="https://github.com/user-attachments/assets/33ce9df7-8e57-4f48-ac5f-3ed2a428c190">
<img width="682" alt="image" src="https://github.com/user-attachments/assets/4a414230-1f99-477d-acb0-a0c91e19b4ae">

@sonnakim for awareness, assigning to @anselmbradford as last modifier of this page.

**Comment by anselmbradford** [comment]
What happened here, is the date picker had a "Types" h2 heading (see https://web.archive.org/web/20240927164342/https://cfpb.github.io/design-system/admin/#/collections/pages/entries/date-picker), which in the markdown is called the variation group name. When I added a second date picker example, this group name was inadvertently indented one level too deep in the markdown, so it was ignored. Decap allowed this, because the variation group name is optional

<img width="442" alt="Screenshot 2024-11-22 at 12 48 22 PM" src="https://github.com/user-attachments/assets/17222b8b-63fc-4ecc-8263-71cdc22d92fc">

This should be fixed for the date picker page now, but because the variation group name is optional it's possible this failure could crop up again in the future across edits to any of the pages.
